src/software/new_geom/util/contains.cpp

```diff
@@ -0,0 +1,45 @@
+#include "software/new_geom/util/contains.h"
+
+#include "software/new_geom/util/collinear.h"
+
+bool contains(const Segment &segment, const Point &point)
```

Comment:
For `Segment` and `Ray`, would it make more sense to have `contains` as class methods? The `Polynomial` classes already have `contains` as methods @scveloso @garethellis0

Reply:
Hmm now that I think about it, I'm leaning towards moving ```contains``` to ```Ray``` and ```Segment```. Even the shape classes have ```contains```.

Reply:
Hmm, I think it makes sense to _maybe_ have `contains(Point)` as a class member, since point is a "clearly smaller" geometry object, but definitely nothing beyond that.

Reply:
Sure, I think it makes sense to just have `contains(Point)` in classes

Reply:
I agree

Reply:
Done in latest